Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -931,36 +940,45 @@ void ExecuteCommand(ReplyType reply_type)
   case DICommand::RequestDiscStatus:
     ERROR_LOG(DVDINTERFACE, "DVDLowRequestDiscStatus");
     DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DVD_LOW_REQUEST_DISC_STATUS);
-    SetHighError(ERROR_INV_CMD);
+    SetDriveError(DriveError::InvalidCommand);
     interrupt_type = DIInterruptType::DEINT;
     break;
   // Wii-exclusive
   case DICommand::RequestRetryNumber:
     ERROR_LOG(DVDINTERFACE, "DVDLowRequestRetryNumber");
     DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DVD_LOW_REQUEST_RETRY_NUMBER);
-    SetHighError(ERROR_INV_CMD);
+    SetDriveError(DriveError::InvalidCommand);
     interrupt_type = DIInterruptType::DEINT;
     break;
   // Wii-exclusive
   case DICommand::SetMaximumRotation:
     ERROR_LOG(DVDINTERFACE, "DVDLowSetMaximumRotation");
-    SetHighError(ERROR_INV_CMD);
+    SetDriveError(DriveError::InvalidCommand);
     interrupt_type = DIInterruptType::DEINT;
     break;
   // Wii-exclusive
   case DICommand::SerMeasControl:
     ERROR_LOG(DVDINTERFACE, "DVDLowSerMeasControl");
     DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DVD_LOW_SER_MEAS_CONTROL);
-    SetHighError(ERROR_INV_CMD);
+    SetDriveError(DriveError::InvalidCommand);
     interrupt_type = DIInterruptType::DEINT;
     break;
 
   // Used by both GC and Wii
   case DICommand::RequestError:
-    INFO_LOG(DVDINTERFACE, "Requesting error... (0x%08x)", s_error_code);
-    s_DIIMMBUF = s_error_code;
-    SetHighError(0);
+  {
+    u8 drive_state;
```

Comment:
This should probably be a `u32`, otherwise the assignment to `result` is intermixing values of different signedness (`drive_state << 24` produces an `int`, not an unsigned value).

Reply:
Changed.  I still have `drive_state = static_cast<u8>(s_drive_state) - 1` though; should that be changed to `u32` too or is `u8` fine?  (the enum itself is u8, and also it's already been checked that s_drive_state is not Ready i.e. 0 so that won't underflow).

Reply:
Changing the cast over to `u32` would likely be better for consistency, though I'm not hard pressed about it.

Reply:
According to [cppreference](https://en.cppreference.com/w/cpp/language/static_cast):

> 7) [Scoped enumeration](https://en.cppreference.com/w/cpp/language/enum)(C++11) type can be converted to an integer or floating-point type.
> <table><tr><td>When the target type is <code>bool</code> (possibly cv-qualified), the result is <code>false</code> if the original value is zero and <code>true</code> for all other values. For the remaining integral types, the result is  the value of the enum if it can be represented by the target type and  unspecified otherwise</td><td>(until C++20)</td></tr>
> <tr><td>The result is the same as <a href="https://en.cppreference.com/w/cpp/language/implicit_conversion">implicit conversion</a> from the enum's underlying type to the destination type.</td><td>(since C++20)</td></tr></table>

which I'm reading as the cast to `u8` followed by implicit conversion to `u32` is the same as just casting to `u32`.  Changed.